app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java

```diff
@@ -105,15 +105,20 @@ public void onRestoreInstanceState(Parcelable state) {
     /**
      * This is a hack, when the SavedState class is unmarshalled a "ClassNotFoundException"
      * will be thrown (the actual class not found is "android.support.design.widget.NavigationView$SavedState")
-     * and I seem to only be able to replicate on Marshmallow.
+     * and I seem to only be able to replicate on Marshmallow (someone else replicated in N through O_MR1
+     * see https://github.com/TeamAmaze/AmazeFileManager/issues/1400#issuecomment-413086603).
      * Trying to find the class and returning false if Class.forName() throws "ClassNotFoundException"
      * doesn't work because the class seems to have been loaded with the current loader
      * (not the one the unmarshaller uses); of course I have no idea of what any of this means so
      * I could be wrong.
      * For the crash see https://github.com/TeamAmaze/AmazeFileManager/issues/1101.
      */
     public boolean doesNavigationViewSavedStateExist() {
-        return Build.VERSION.SDK_INT != Build.VERSION_CODES.M;
+        return Build.VERSION.SDK_INT != Build.VERSION_CODES.M
+                && Build.VERSION.SDK_INT != Build.VERSION_CODES.N
+                && Build.VERSION.SDK_INT != Build.VERSION_CODES.N_MR1
+                && Build.VERSION.SDK_INT != Build.VERSION_CODES.O
+                && Build.VERSION.SDK_INT != Build.VERSION_CODES.O_MR1;
```

Comment:
If we're certain `Build.VERSION_CODES` is a series of consequential integers without skipping, why not just do
```
return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
     && Build.VERSION.SDK_INT <= Build.VERSION_CODES.O_MR1;
```

Reply:
More like ````return Build.VERSION.SDK_INT<Build.VERSION_CODES.M````

Reply:
Right, **is not** between Android M and Android O MR1.

Reply:
> More like `return Build.VERSION.SDK_INT<Build.VERSION_CODES.M`

Remember that we don't know if future versions will have this issue.

Anyway, fixed.

Reply:
@EmmanuelMess I think you directly picked @TranceLove 's code without negating it. Please check once.